libraries/chain/include/eosio/chain/webassembly/interface.hpp

```diff
@@ -0,0 +1,581 @@
+         REGISTER_LEGACY_HOST_FUNCTION(db_idx_double_lowerbound, is_nan_check);
+         REGISTER_LEGACY_HOST_FUNCTION(db_idx_double_upperbound, is_nan_check);
+         REGISTER_LEGACY_HOST_FUNCTION(db_idx_double_end);
+         REGISTER_HOST_FUNCTION(db_idx_double_next);
+         REGISTER_HOST_FUNCTION(db_idx_double_previous);
+
+
+         // long double secondary index api
+         int32_t db_idx_long_double_store(uint64_t, uint64_t, uint64_t, uint64_t, const float128_t&);
+         void db_idx_long_double_update(int32_t, uint64_t, const float128_t&);
+         void db_idx_long_double_remove(int32_t);
+         int32_t db_idx_long_double_find_secondary(uint64_t, uint64_t, uint64_t, legacy_ptr<const float128_t>, legacy_ptr<uint64_t>);
+         int32_t db_idx_long_double_find_primary(uint64_t, uint64_t, uint64_t, float128_t&, uint64_t);
+         int32_t db_idx_long_double_lowerbound(uint64_t, uint64_t, uint64_t, legacy_ptr<float128_t, 8>, legacy_ptr<uint64_t, 8>);
+         int32_t db_idx_long_double_upperbound(uint64_t, uint64_t, uint64_t, legacy_ptr<float128_t, 8>,  legacy_ptr<uint64_t, 8>);
+         int32_t db_idx_long_double_end(uint64_t, uint64_t, uint64_t);
+         int32_t db_idx_long_double_next(int32_t, uint64_t&);
+         int32_t db_idx_long_double_previous(int32_t, uint64_t&);
+
+         REGISTER_HOST_FUNCTION(db_idx_long_double_store, is_nan_check);
+         REGISTER_HOST_FUNCTION(db_idx_long_double_update, is_nan_check);
+         REGISTER_LEGACY_HOST_FUNCTION(db_idx_long_double_remove);
+         REGISTER_LEGACY_HOST_FUNCTION(db_idx_long_double_find_secondary, is_nan_check);
+         REGISTER_HOST_FUNCTION(db_idx_long_double_find_primary);
+         REGISTER_LEGACY_HOST_FUNCTION(db_idx_long_double_lowerbound, is_nan_check);
+         REGISTER_LEGACY_HOST_FUNCTION(db_idx_long_double_upperbound, is_nan_check);
+         REGISTER_LEGACY_HOST_FUNCTION(db_idx_long_double_end);
+         REGISTER_HOST_FUNCTION(db_idx_long_double_next);
+         REGISTER_HOST_FUNCTION(db_idx_long_double_previous);
+
+         // memory api
+         char* memcpy(unvalidated_ptr<char>, unvalidated_ptr<const char>, wasm_size_t) const;
+         char* memmove(unvalidated_ptr<char>, unvalidated_ptr<const char>, wasm_size_t) const;
+         int32_t memcmp(unvalidated_ptr<const char>, unvalidated_ptr<const char>, wasm_size_t) const;
+         char* memset(unvalidated_ptr<char>, int32_t, wasm_size_t) const;
+
+         REGISTER_LEGACY_CF_HOST_FUNCTION(memcpy);
+         REGISTER_LEGACY_CF_HOST_FUNCTION(memmove);
+         REGISTER_LEGACY_CF_HOST_FUNCTION(memcmp);
+         REGISTER_LEGACY_CF_HOST_FUNCTION(memset);
+
+         // transaction api
+         void send_inline(legacy_array_ptr<char>);
+         void send_context_free_inline(legacy_array_ptr<char>);
+         void send_deferred(legacy_ptr<const uint128_t>, account_name, legacy_array_ptr<char>, uint32_t);
+         bool cancel_deferred(legacy_ptr<const uint128_t>);
```

Comment:
The first argument of `send_deferred` and `cancel_deferred` used to be a a `const uint128_t&` but is now a pointer to a `const uint128_t`.

Reply:
This doesn't change the behavior.  I can just rename legacy_ptr to legacy_ref.